src/main/java/de/tum/in/www1/artemis/service/QuizExerciseImportService.java

```diff
@@ -149,6 +154,36 @@ else if (quizQuestion instanceof ShortAnswerQuestion saQuestion) {
         newExercise.setQuizQuestions(importedExercise.getQuizQuestions());
     }
 
+    /**
+     * Checks whether an alleged backgroundFilePath is actually one. In case it isn't, a "Background File Path is not valid!"-IllegalArgumentException is thrown.
+     *
+     * @param backgroundFilePath Path to be checked
+     */
+    public static void checkIfBackgroundFilePathIsABackgroundFilePathElseThrow(String backgroundFilePath) {
+        /*
+         * FileService.copyExistingFileToTarget(String, String, Long) only acts,
+         * when backgroundFilePath != null && !backgroundFilePath.contains("files/temp")
+         * => No need to check if backgroundFilePath == null || backgroundFilePath.contains("files/temp")
+         */
+        if (backgroundFilePath != null && !backgroundFilePath.contains("files/temp")) {
+            // Substitutes all escape sequences in the backgroundFilePath with their standard elements
+            String unescapedBackgroundFilePath = StringEscapeUtils.unescapeJava(backgroundFilePath);
+            // Removes redundant elements (e.g. ../ or ./) from the backgroundFilePath
+            Path normalisedBackgroundFilePath = Paths.get(unescapedBackgroundFilePath).normalize();
+            // Resolve the backgroundFilePath on the empty path in order to receive an absolute path
+            Path absoluteBackgroundFilePath = Path.of("").resolve(normalisedBackgroundFilePath);
+            /*
+             * Check whether the absolute backgroundFilePath without any escape sequences
+             * or redundant elements starts with "/api/files/drag-and-drop/backgrounds"
+             * (the standard actual path for drag-and-drop background images)
+             */
+            if (!absoluteBackgroundFilePath.startsWith("/api/files/drag-and-drop/backgrounds")) {
```

Comment:
Are you sure that an `absolute` path will ever resolve to that? But it's challenging to talk about anyway since it's not a file path, as mentioned.

The test covers it, and the manual tests work, too. Do you have any idea why this works? Since it doesn't look very absolute to me.

Reply:
Removed absolute path in [69a0c8b408b16a8e68ace79dfa4cee7927e48db0](https://github.com/ls1intum/Artemis/pull/7170/commits/69a0c8b408b16a8e68ace79dfa4cee7927e48db0).